estuary/utils/recents.py

```diff
@@ -38,9 +50,10 @@ def get_recent_nodes():
         ).format(label=label, time_property=time_property)
         results, _ = db.cypher_query(query)
         for result in results:
-            node_results = final_result.setdefault(label, [])
+            node_results = final_result_data.setdefault(label, [])
             # result is always a list of a single node
             node = inflate_node(result[0])
             node_results.append(node.serialized)
+            id_dict[node.__label__] = node.unique_id_property
```

Comment:
As an efficiency improvement, you could check if the `id_dict[node.__label__]` is already set before setting it since `unique_id_property` is a property method that executes code to determine the value, and it's a bit wasteful to keep running that over and over again.